nfqueue_gteq_1.12.go

```diff
@@ -145,13 +145,19 @@ func (nfqueue *Nfqueue) socketCallback(ctx context.Context, fn HookFunc, seq uin
 		}
 		replys, err := nfqueue.Con.Receive()
 		if err != nil {
-			if opError, ok := err.(*netlink.OpError); ok {
-				if opError.Timeout() || opError.Temporary() {
-					continue
+			if errfn != nil {
```

Comment:
With the use of an anonymous function in `Register()` we don't need to check against nil here and can directly use 

```
if ret := errfn(err); ret != 0 {
    return
}
```
```
func (nfqueue *Nfqueue) Register(ctx context.Context, fn HookFunc) error {
	return nfqueue.RegisterWithErrorFunc(ctx, fn, func(e error) int {
		if opError, ok := err.(*netlink.OpError); ok {
			if opError.Timeout() || opError.Temporary() {
				return 0
			}
		}
		nfqueue.logger.Printf("Could not receive message: %v\n", err)
		return 1
	})
}
```

Reply:
Good idea! I updated and force-pushed the PR.